### DISCUSSION TITLE
0.37.1-a2

### DISCUSSION THREAD

**Discussion by rmorshea** [idom-team/idom#698] [Announcements]
## What's Changed
* set target value onchange client-side by @rmorshea in https://github.com/idom-team/idom/pull/694
* actually fix #684 by @rmorshea in https://github.com/idom-team/idom/pull/696
* fix flicker with observedValues buffer by @rmorshea in https://github.com/idom-team/idom/pull/697

**Full Changelog**: https://github.com/idom-team/idom/compare/0.37.0...0.37.1-a2

<hr /><em>This discussion was created from the release <a href='https://github.com/idom-team/idom/releases/tag/0.37.1-a2'>0.37.1-a2</a>.</em>

**Comment by rmorshea** [comment]
@acivitillo, can you confirm whether or not this addresses the input issue you've been observing?

**Reply by acivitillo** [reply]
I can confirm the `value` setting now works on `input` element, I need to review further with the team the other issue with `select` elements not changing in the client between component changes

**Reply by acivitillo** [reply]
This works now:

```python
@component
def Input(
    set_value: Callable,
    label: str = "",
    type: str = "text",
    placeholder: str = "Write here the",
    _class: str = class_str,
    value: Any = "",
):

    return html.input(
        {
            "type": type,
            "placeholder": f"{placeholder} {label}",
            "onChange": lambda event: set_value(event["target"]["value"]),
            "class": _class,
            "value": value,
        }
    )
```

**Reply by rmorshea** [reply]
The `select` element is probably the same issue and should be fixed too. `input`, `select`, and `textarea` all support the `onChange` event and have a `value` attribute.
